src/lib/interactionManager/authorizationFlow.ts

```diff
@@ -0,0 +1,56 @@
+import { Flow } from './flow'
+import {
+  FlowType,
+  AuthorizationResponse,
+  AuthorizationRequest,
+  AuthorizationType,
+  AuthorizationFlowState,
+} from './types'
+import { isAuthorizationRequest, isAuthorizationResponse } from './guards'
+
+export class AuthorizationFlow extends Flow<
+  AuthorizationResponse | AuthorizationRequest
+> {
+  public type = FlowType.Authorization
+  public state: AuthorizationFlowState = {
+    description: '',
+    imageURL: '',
+    action: '',
+  }
+
+  public handleInteractionToken(
+    token: AuthorizationRequest | AuthorizationResponse,
+    interactionType: string,
+  ) {
+    switch (interactionType) {
+      case AuthorizationType.AuthorizationRequest:
+        if (isAuthorizationRequest(token, interactionType))
+          return this.consumeAuthorizationRequest(token)
+      case AuthorizationType.AuthorizationResponse:
+        if (isAuthorizationResponse(token, interactionType))
+          return this.consumeAuthorizationResponse(token)
+      default:
+        throw new Error('Interaction type not found')
+    }
+  }
+
+  public async consumeAuthorizationRequest(request: AuthorizationRequest) {
+    if (!this.state.description) this.state.description = request.description
```

Comment:
I think we don't have to perform all these check here.
I.e. I suggest we check if any of the state values are set, and if yes, return `false`, since we do not intend for this method to be called twice. Once we have set the values, we can return `true`.